### PR TITLE
Documented annotation volume.kubernetes.io/storage-provisioner

### PR DESCRIPTION
**Pull request by Kartik494**
Fixes: #30807

### CODE REVIEW

**Comment by sftim** [comment]
@kubernetes/sig-storage-pr-reviews does this look right? Anything more to add?

**Comment by jsafrane** [comment]
lgtm from storage POV

File: content/en/docs/reference/labels-annotations-taints.md [hunk 1]
```diff
@@ -159,6 +159,20 @@ The scheduler (through the _VolumeZonePredicate_ predicate) also will ensure tha
 If `PersistentVolumeLabel` does not support automatic labeling of your PersistentVolumes, you should consider
 adding the labels manually (or adding support for `PersistentVolumeLabel`). With `PersistentVolumeLabel`, the scheduler prevents Pods from mounting volumes in a different zone. If your infrastructure doesn't have this constraint, you don't need to add the zone labels to the volumes at all.
 
+## volume.beta.kubernetes.io/storage-provisioner (deprecated)
+
+Example: `volume.beta.kubernetes.io/storage-provisioner: k8s.io/minikube-hostpath`
+
+Used on: PVC
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints.md:166]:
```suggestion
Used on: PersistentVolumeClaim
```

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks for this.

One thing to fix @Kartik494 - use the type name that you would see in a manifest.

File: content/en/docs/reference/labels-annotations-taints.md [hunk 2]
```diff
@@ -159,6 +159,20 @@ The scheduler (through the _VolumeZonePredicate_ predicate) also will ensure tha
...
+
+This annotation has been deprecated.
+
+## volume.kubernetes.io/storage-provisioner
+
+Used on: PVC
```
[the remark this answers is not in the document]
Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints.md]: @sftim Done. PTAL!

**Review by sftim — APPROVED** [review]
Thanks

/lgtm
/approve

**Comment by sftim** [comment]
/retitle Document annotation volume.kubernetes.io/storage-provisioner

**Pull request merged**